### PR TITLE
Api 40298 ews bd upload refactor

### PR DESCRIPTION
**Pull request by mchristiansonVA**
## Summary

- *This work is behind a feature toggle (flipper): YES*
- Adds new class PoaDocumentService to move document-specific logic for POA uploads out of the BD class, to reduce the concerns handled by the BD class to simply handling the BD upload given a request body.

## Related issue(s)

[API-40298](https://jira.devops.va.gov/browse/API-40298)

## Testing done

- [x] *New code is covered by unit tests*
- Prior EWS upload logic used ClaimsApi::BD service methods to handle logic specific to the doc type and generate the appropriate upload body for submission to BD /upload.
- New code uses EvidenceWaiverDocumentService and DocumentServiceBase classes to handle logic specific to the L705 doc type and generate the appropriate upload body for submission to BD /upload, using new ClaimsApi::BD service method upload_document to simply submit to BD /upload.
- New logic requires the following flipper to be enabled:
Flipper.enable :claims_api_ews_uploads_bd_refactor
- Tested EWS submissions and confirmed BD /upload success.

## What areas of the site does it impact?
EWS submissions

## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

### CODE REVIEW

**Review by FonzMP — APPROVED** [review]
:shipit:

File: modules/claims_api/spec/sidekiq/evidence_waiver_builder_job_spec.rb [hunk 1]
```diff
@@ -44,4 +44,16 @@
       end
     end
   end
+
+  context 'when the claims_api_ews_uploads_bd_refactor BD refactor feature flag is enabled' do
```
Comment (stevenjcumming) [inline modules/claims_api/spec/sidekiq/evidence_waiver_builder_job_spec.rb:48]: Please add `context 'when claims_api_ews_uploads_bd_refactor is disabled'`

Comment (mchristiansonVA) [reply modules/claims_api/spec/sidekiq/evidence_waiver_builder_job_spec.rb:48]: Context added for feature disabled.

**Review by rockwellwindsor-va — APPROVED** [review]
Ran this workflow from beginning to end and it all worked as expected, LGTM!

Screenshot 2024-10-24 at 7 33 02 AM

**Review by rockwellwindsor-va — APPROVED** [review]
Still LGTM!

**Pull request merged**